File: Python_Books/算法图解/grokking_algorithms_COPY/03_recursion/c/02_greet.c
#include <stdio.h>

void greet2(char *name) {
	printf("how are you, %s?\n", name);
}

void bye() {
	printf("ok bye!\n");
}

void greet(char *name) {
	printf("hello, %s!\n", name);
	greet2(name);
	printf("getting ready to say bye...\n");
	bye();
}

int main(void) {
	
	greet("adit");

	return 0;
}

File: Python_Books/算法图解/grokking_algorithms_COPY/03_recursion/javascript/02_greet.js
function greet2(name) {
  console.log('how are you, ' + name + '?');
}

function bye() {
  console.log('ok bye!');
}

function greet(name) {
  console.log('hello, ' + name + '!');
  greet2(name);
  console.log('getting ready to say bye...');
  bye();
}

greet('adit');
